File: tpDEMO/app/src/main/java/com/wcch/android/utils/SPUtils.java
package com.wcch.android.utils;

import android.content.Context;
import android.content.SharedPreferences;

import static android.content.Context.MODE_PRIVATE;

public class SPUtils {


    private static Context mContext;
    private static SharedPreferences mSharedPreferences;


    private static synchronized SharedPreferences getPreferneces() {
        if (mSharedPreferences == null) {
            mSharedPreferences = mContext.getSharedPreferences("projection", MODE_PRIVATE);
        }
        return mSharedPreferences;
    }

    public static void initSP(Context context) {
        mContext = context;
    }

    //保存字符串
    public static void putString(String key, String value) {
        getPreferneces().edit().putString(key, value).commit();
    }
    //读取字符串
    public static String getString(String key) {
        return getPreferneces().getString(key, null);
    }

    //读取字符串
    public static String getString(String key, String def) {
        return getPreferneces().getString(key, def);
    }
    //保存整型值
    public static void putInt(String key, int value) {
        getPreferneces().edit().putInt(key, value).commit();
    }
    //读取整型值
    public static int getInt(String key) {
        return getPreferneces().getInt(key, -1);
    }
    public static int getInt(String key, int defValue) {
        return getPreferneces().getInt(key, defValue);
    }
    //保存布尔值
    public static void putBool(String key, Boolean value) {
        getPreferneces().edit().putBoolean(key, value).commit();
    }
    public static void putLong(String key, long value) {
        getPreferneces().edit().putLong(key, value).commit();
    }
    public static long getResumeSize(String key) {
        return getPreferneces().getLong(key, 10);
    }

    public static long getLong(String key) {
        return getPreferneces().getLong(key, 0);
    }
    //读取布尔值
    public static boolean getBool(String key, boolean defValue) {
        return getPreferneces().getBoolean(key, defValue);
    }
    //移除字段
    public static void removeString(String key) {
        getPreferneces().edit().remove(key).commit();
    }
    //清空保存在默认SharePreference下的所有数据
    private static void clear() {
        getPreferneces().edit().clear().commit();
    }

    public static void setLogStatus(boolean value) {
        mSharedPreferences.edit().putBoolean("CAN_PRINT_LOG",value).commit();
    }

    public static boolean getLogStatus() {
        return mSharedPreferences.getBoolean("CAN_PRINT_LOG",false);
    }
}

File: tpSoftDemo/app/src/main/java/com/wcch/android/soft/util/SpUtils.kt
package com.wcch.android.soft.util

import android.content.SharedPreferences
import com.wcch.android.App

/**
 * @author markLiu
 * @title SpUtils
 * @time 2021/2/2 15:21
 * @description sp存储
 */
class SpUtils {
    var sp: SharedPreferences? = null
        get() {
            if (field == null) field = App.getInstance().applicationContext!!.getSharedPreferences("set", 0)
            return field
        }
        private set

    fun containsKey(key: String?): Boolean {
        return sp!!.contains(key)
    }

    fun save(key: String?, value: Int) {
        sp!!.edit().putInt(key, value).apply()
    }

    fun getInt(key: String?): Int {
        return sp!!.getInt(key, -1)
    }

    fun save(key: String?, value: String?) {
        sp!!.edit().putString(key, value).apply()
    }

    fun getString(key: String?): String? {
        return sp!!.getString(key, "")
    }

    fun save(key: String?, value: Boolean) {
        sp!!.edit().putBoolean(key, value).apply()
    }

    fun getBoolean(key: String?): Boolean {
        return sp!!.getBoolean(key, false)
    }

    fun getBoolean(key: String?, defValue: Boolean): Boolean {
        return sp!!.getBoolean(key, defValue)
    }

    fun save(key: String?, value: Long) {
        sp!!.edit().putLong(key, value).apply()
    }

    fun getLong(key: String?): Long {
        return sp!!.getLong(key, -1)
    }

    companion object {
        @JvmStatic
        var instance: SpUtils? = null
            get() {
                if (field == null) field = SpUtils()
                return field
            }
            private set
    }
}